tracing-subscriber/src/fmt/writer.rs

```diff
@@ -54,6 +54,31 @@ where
     }
 }
 
+/// Prints text using the standard library's macro `print!`. This writer allows stdout logs to be
+/// correctly captured by commands such as `cargo test`.
+#[derive(Debug)]
+pub struct TestWriter;
```

Comment:
Since this struct has no fields, it can be constructed in a downstream crate as
```rust
let writer = TestWriter;
```
or similar.

If we were to add any fields to it in the future, though, it would no longer be possible to construct it this way, which would be a breaking change.

Therefore, for future-proofing reasons, it would be better to [add a private field, just to prevent it from being constructed in this way](https://rust-lang.github.io/api-guidelines/future-proofing.html#structs-have-private-fields-c-struct-private), and add a `new` function and/or `Default` implementation for constructing it. That way, if we need to add fields to this struct later (such as if we decide to implement capturing for threads spawned inside a test as well as the main test thread, as discussed here https://github.com/tokio-rs/tracing/issues/638#issuecomment-600747298) we can do so without a breaking change.

We could add an unused, private `()` field to the struct, like this:

```suggestion
#[derive(Default)]
pub struct TestWriter {
    _p: (),
}
```